### PR TITLE
added disconnect listener

### LINKED ISSUE #38: [Help] How to attach a callback to a disconnect event?
How do I detect when a device is disconnected? The scrcpy program is able to tell when it gets disconnected and closes the stream with a log `WARN: Device disconnected`. But when I disconnect when using the python client, the frame feed just stops and there are no logs. I tried using `adb.wait_for` but it is a blocking function.

### PR DESCRIPTION
**Pull request by FieryRMS**
fixes #38 
`self.__video_socket.recv` returns an empty byte string when device gets disconnected.
I also added a new event `EVENT_DISCONNECT` that is called when disconnect happens.

https://github.com/leng-yue/py-scrcpy-client/blob/f94749b9d187e264e2ef394a9b00a2228572f79f/scrcpy/core.py#L227

### CODE REVIEW

File: scrcpy/core.py [hunk 1]
```diff
@@ -225,6 +230,8 @@ def __stream_loop(self) -> None:
         while self.alive:
             try:
                 raw_h264 = self.__video_socket.recv(0x10000)
+                if(raw_h264==b''):
```
Comment (leng-yue) [inline scrcpy/core.py:233]: lint issue

File: scrcpy/core.py [hunk 2]
```diff
@@ -225,6 +230,8 @@ def __stream_loop(self) -> None:
...
+                    raise ConnectionAbortedError
```
Comment (leng-yue) [inline scrcpy/core.py:234]: You can raise the `ConnectionError("Video stream is disconnected")`

File: scrcpy/core.py [hunk 3]
```diff
@@ -239,6 +246,11 @@ def __stream_loop(self) -> None:
                 time.sleep(0.01)
                 if not self.block_frame:
                     self.__send_to_listeners(EVENT_FRAME, None)
+            except ConnectionAbortedError as e:
+                if self.alive:
+                    self.__send_to_listeners(EVENT_DISCONNECT)
+                    self.stop()
+                    raise e
             except OSError as e:  # Socket Closed
```
Comment (leng-yue) [inline scrcpy/core.py:254]: Then you can `expect (ConnectionError, OSError) as e`

File: scrcpy/core.py [hunk 4]
```diff
@@ -239,6 +246,11 @@ def __stream_loop(self) -> None:
...
                 if self.alive:
                     self.stop()
```
Comment (leng-yue) [inline scrcpy/core.py:256]: And send the event to the listener here since OSError also means the device is physically disconnected.

**Comment by FieryRMS** [comment]
I really need to learn how to use github properly :')

**Pull request merged**